### PR TITLE
libiconv: Update SPDX license identifier and set license based on version

### PR DESCRIPTION
**Pull request by jwillikers**
libiconv 1.17 switched to the LGPL-2.1 from the LGPL-2.0 license. The Conan package previously didn't express the license for older versions correctly.

Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libiconv/all/conanfile.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 class LibiconvConan(ConanFile):
     name = "libiconv"
     description = "Convert text to and from Unicode"
-    license = "LGPL-2.1"
+    license = "LGPL-2.1-or-later"
```
Comment (prince-chrismc) [inline recipes/libiconv/all/conanfile.py:27]: https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=blob;f=include/iconv.h.in;h=62f7b351a3907ec85df39679b2803460425487b0;hb=HEAD#l7 👍

File: recipes/libiconv/all/conanfile.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ def configure(self):
             self.options.rm_safe("fPIC")
         self.settings.rm_safe("compiler.libcxx")
         self.settings.rm_safe("compiler.cppstd")
+        if Version(self.version) < "1.17":
+            self.license = "LGPL-2.0-or-later"
```
Comment (prince-chrismc) [inline recipes/libiconv/all/conanfile.py:67]:
```suggestion
        if Version(self.version) >= "1.17":
            self.license = "LGPL-2.1-or-later"
```

**Review by prince-chrismc — COMMENTED** [review]
Small suggestion to let the "base recipe" list all the licenses and then bump up the "min or later" starting at the version (simple way but there could be an else if you'd like)

changing it in configure only impacts the package not the recipe inspection AFAIK so this will allow all the command to have the most information

File: recipes/libiconv/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/libiconv/all/conanfile.py]: Sure thing!

File: recipes/libiconv/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/libiconv/all/conanfile.py]: Done. I also added the `else` clause.

**Comment by jwillikers** [comment]
@prince-chrismc Thanks, that's helpful to get right. I have some things check license compliance in the `validate` method, so was mostly concerned with that. It's helpful to know how to get the inspection to show the relevant licenses, too.

**Comment by prince-chrismc** [comment]
I like to use `conan info recipes/libiconv/all/conanfile.py` 

```sh
conanfile.py (libiconv/None)
    ID: 299aaabcb5fb1385c096ccf2c550a653c4438173
    BuildID: None
    Context: host
    URL: https://github.com/conan-io/conan-center-index
    Homepage: https://www.gnu.org/software/libiconv/
    License: LGPL-2.1
    Description: Convert text to and from Unicode
    Topics: iconv, text, encoding, locale, unicode, conversion
    Provides: libiconv
```

The "none" at the top is what usually gets me since it's not been processed.

I do not have ur branch on this PC but this will have both with our changes.

Then you can do 

```sh
$ conan inspect quirc/1.1@
name: quirc
version: 1.1
url: https://github.com/conan-io/conan-center-index
homepage: https://github.com/dlbeer/quirc
license: ISC
```

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
Well, just use shared library and you will be okay!

**Pull request merged**